contracts/v3/Vault.sol

```diff
@@ -237,7 +237,12 @@ contract Vault is VaultToken, IVault {
         override
         checkToken(_output)
     {
-        uint256 _amount = (balance().mul(_shares)).div(totalSupply());
+        if (totalSupply() > 0) {
+            uint256 _amount = (balance().mul(_shares)).div(totalSupply());
+        }
+        else {
+            uint256 _amount = 0;
+        }
         _burn(msg.sender, _shares);
```

Comment:
For the case when `_totalSupply` is `0`, the `_burn` will decrease the total supply, which will revert because of safemath. As long as `_shares > 0`.